### PR TITLE
18. As a user, I want to log out and see my token from the navigation bar

### PR DESCRIPTION
**Pull request by jessicasalbert**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

This code finishes the functionality of our nav bar with the addition of a log out and token share feature. The log out button removes the token currently saved in local storage, and redirects the user back to the home page. The action is confirmed with a modal. The token share button prompts a modal to display the current users token so it can be used to share the shopping list. 

## Related Issue

Closes #18

## Acceptance Criteria

- "log out" -> removes token from local storage and brings user back to home page once they click on the icon
- "share token" -> using sweet alerts (which is already added as a dependency), show the user their token with a message once they click on the icon (i.e. “Share this token with friends or use it yourself to access this list again: mark marks swipe”

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Buttons were already preset in the UI - PR just implements the functionality

### After

[Image from Gyazo](https://gyazo.com/e046cd3971f4ca799bb1ccc216d37774)

[Image from Gyazo](https://gyazo.com/e0fda654f232621967c4fc186da4b52d)

## Testing Steps / QA Criteria

- Log into a list 
- Click the "share" icon on the nav bar - ensure that the token displayed in the modal matches the token saved in local storage
- Click the "logout" icon on the nav bar - a confirmation modal should appear
- Upon confirming logout, user should be redirected to home page

### CODE REVIEW

File: src/components/Navbar/Navbar.js [hunk 1]
```diff
@@ -8,9 +8,41 @@ import { ReactComponent as AddItem } from "../../img/add_item.svg";
...
+    });
+  };
+
+  const logoutHandler = () => {
+    swal({
+      title: "Do you want to log out?",
+      icon: "warning",
+      buttons: true,
+      dangerMode: true,
+    }).then((willLogout) => {
```
Comment (thekidnamedkd) [inline src/components/Navbar/Navbar.js]: This is great! Just thinking from a UX perspective about the copy: maybe we should be extra explicit about "write down this three-word token phrase to share access with friends or reaccess this list yourself" and "Do you want to eject this list and access another?" or something. Really nice work on the code!

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -35,10 +67,10 @@ export default function Navbar(props) {
             <Link className="navLink" to="/AddItem">
               <AddItem />
             </Link>
-            <Link className="navLink" to="/">
+            <Link className="navLink" to="/" onClick={logoutHandler}>
```
Comment (thekidnamedkd) [inline src/components/Navbar/Navbar.js]: Akin to the previous comment maybe this becomes `{clearTokenHandler}` or something to make this flow clearer on the developer side as well.

**Review by thekidnamedkd — APPROVED** [review]
Nice job! Sweet alert strikes again. Great job with lining all this up. I think we have an effective way to swap lists around now.

File: src/components/Navbar/Navbar.js [hunk 3]
```diff
@@ -8,9 +8,41 @@ import { ReactComponent as AddItem } from "../../img/add_item.svg";
...
+      dangerMode: true,
+    }).then((willLogout) => {
+      if (willLogout) {
+        localStorage.removeItem("token");
+        props.setAuth(false);
+        history.push("/");
+        swal({
+          text: "You have successfully logged out",
+          dangerMode: true,
+          icon: "success",
```
Comment (wlcreate) [inline src/components/Navbar/Navbar.js]: I like that there's a modal to show that the user successfully logged out, but I don't think we need to have the button present for them to close it.

**Review by wlcreate — CHANGES_REQUESTED** [review]
Great job handling this issue! In addition to Kevin's comment about the copy I think not having a button for the successful log out would make for better UX.

File: src/components/Navbar/Navbar.js (hunk 1 above)
Comment (jessicasalbert) [reply src/components/Navbar/Navbar.js]: Good idea - I've updated the code!

File: src/components/Navbar/Navbar.js (hunk 3 above)
Comment (jessicasalbert) [reply src/components/Navbar/Navbar.js]: True! I've fixed that in this code. We could consider the same thing for the modal that pops up when deleting an item.

**Review by Maeesha-Rahman — APPROVED** [review]

**Review by wlcreate — APPROVED** [review]
Nice job with the changes! 🥳

**Pull request merged**